components/Form/Checkbox/Checkbox.tsx

```diff
@@ -1,32 +1,39 @@
-import { shape, string, node, number, object, objectOf, oneOfType } from 'prop-types';
 import classNames from 'classnames';
 import { ErrorMessage } from 'formik';
 import { CHECKBOX, CHECKBOX_ERROR } from 'common/constants/testIDs';
 import Alert from 'components/Alert/Alert';
 import Label from 'components/Form/Label/Label';
 import styles from './Checkbox.module.css';
 
-Checkbox.propTypes = {
-  field: shape({
-    name: string.isRequired,
-  }).isRequired,
-  form: shape({
-    // TODO: Resolve why multiselects can end up with touched: { key: array }
-    // see ThemedReactSelect as well
-    // touched: objectOf(bool).isRequired,
-    touched: object.isRequired,
-    errors: objectOf(string),
-  }).isRequired,
-  id: oneOfType([string, number]),
-  label: oneOfType([node, string]).isRequired,
+export type CheckboxPropsType = {
+  /**
+   * Applies a label that to the form input.
+   */
+  label: React.ReactNode | string;
+  /**
+   * Sets the name and value for the input element.
+   */
+  field: {
+    name: string;
+    value?: string;
+  };
+  form: {
+    touched: Record<string, any>;
+    errors?: Record<string, string>;
+  };
```

Comment:
now that we're using TypeScript, we can extend from the actual Formik types!

Reply:
Alternatively, we could not expose those props anymore and leverage the `useField` hook

Reply:
Yeah, so I guess that's kinda my feedback for all of this PR! Choose your path 😂